js/data/feature_tree.js

```diff
@@ -66,15 +66,10 @@ FeatureTree.prototype.query = function(args, callback) {
 };
 
 function geometryContainsPoint(rings, type, p, radius) {
-    if (type === 'Point') {
-        return pointContainsPoint(rings, p, radius);
-    } else if (type === 'LineString') {
-        return lineContainsPoint(rings, p, radius);
-    } else if (type === 'Polygon') {
-        return polyContainsPoint(rings, p) ? true : lineContainsPoint(rings, p, radius);
-    } else {
-        return false;
-    }
+    if (type === 'Point') return pointContainsPoint(rings, p, radius);
```

Comment:
I dislike having condition and statement on the same line because it makes it impossible to set a breakpoint on just the statement. And in this case the code becomes too cramped as well.